services/QuillLMS/app/controllers/teachers/units_controller.rb

```diff
@@ -139,17 +139,31 @@ def destroy
   # :activity_id (in url)
   # :classroom_unit_id
   def score_info
-    completed = ActivitySession.where(
-      classroom_unit_id: params[:classroom_unit_id],
-      activity_id: params[:activity_id],
-      is_final_score: true,
-      visible: true
-    )
+    classroom_unit = ClassroomUnit.find(params[:classroom_unit_id])
+    cache_groups = {
+      activity: params[:activity_id]
+    }
+
+    json = current_user.classroom_unit_cache(classroom_unit, key: 'units.score_info', groups: cache_groups) do
```

Comment:
For consistency with the other caching PRs, maybe make `key: 'teachers.units.score_info'`?

Reply:
Yeah, good idea.